proposals/0000-linear-types.rst

```diff
@@ -519,7 +526,8 @@ has the consequence of consuming all the resources in the closure of
 ``v`` making it safe to use the value many times or not at all. But
 newtypes convert ``case`` into a cast, hence the closure is never
 consumed. So ``newtype`` must not accept non-linear arrow with
-``-XLinearTypes``. This is not backward compatible.
+``-XLinearTypes``. These are interpreted as linear ``newtype``-s and a
+warning is emitted (see Specification above).
```

Comment:
I'd treat it as an error. The compiler shouldn't change the meaning that the programmer intended.

Reply:
This is a small sacrifice that I'm willing to do on the altar of backwards compatibility (it's explained in the Specification section).

Reply:
If a module is going to be updated to use `LinearTypes`, it surely can update an unrestricted newtype too, right? I don't see what costs are being saved by making it a warning.

Reply:
Oh I see the objection. Mmm…

This affects every newtype-gadt definition. Even if `-XLinearType` is not used. Hence the compatibility issue. On the other hand, if we have a warning for people not using linear types, then we're basically making `-XLinearTypes` necessary to write newtype-gadts. Mmmm. I'm really not sure what the best solution is here.